tests/unit/certtool.py

```diff
@@ -0,0 +1,100 @@
+import datetime
+from cryptography.hazmat.primitives import hashes, serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography import x509
+from cryptography.x509 import NameAttribute
+from cryptography.x509.base import Certificate
+from cryptography.x509.oid import NameOID
+
+# Class to generate a test certificate chain
+# https://cryptography.io/en/latest/x509/tutorial/
+class CertGen:
+    def __init__(self):
+        self.init_root_cert()
+        self.init_postgresql_cert()
+        self.dump_to_storage()
+
+    def dump_to_storage(self):
+        self.cert_map = {}
+        self._dump_cert(self.root_cert, "root.crt")
```

Comment:
Can you perhaps use temp files for both files, and pass the paths / handles on to the consumer?
Currently, there might be leftovers when you run unit tests locally (so outside of a container).

Reply:
The temporary files created by database.py are supposed to stay after the python program exits. The consumer can later extract the paths from the jdbc URL and delete them.

The unit test does not create additional temporary files. It stores the expected text and binary payloads in latin-1 strings in a dictionary. File-like names are only to distinguish different kinds of certificates in `cert_map`. latin-1 encoding is chosen because most of the time I compare PEM payloads, but it's good for binary too.

If the check succeeds the unit test deletes the temporary files created by database.py. If it fails, some files will remain, but maybe it's for the better as it makes troubleshooting easier. So I don't want to add the additional cleanup here.